flink-runtime/src/main/java/org/apache/flink/runtime/io/network/ShuffleEnvironment.java

```diff
@@ -0,0 +1,163 @@
+ * externally, ouside of the task thread, when the producer becomes known.
+ */
+public interface ShuffleEnvironment {
+
+	/**
+	 * Starts the internal related services upon {@link TaskExecutor}'s startup.
+	 *
+	 * @return a port to connect to the task executor for shuffle data exchange, -1 if only local connection is possible.
+	 */
+	int start() throws IOException;
+
+	/**
+	 * Shutdown the internal related services upon {@link TaskExecutor}'s stop.
+	 */
+	void shutdown();
+
+	boolean isShutdown();
+
+	/**
+	 * Factory method for the task's {@link ResultPartitionWriter}s.
+	 *
+	 * @param taskName string name of the task used for logs
+	 * @param executionAttemptID execution attempt id of the task
+	 * @param resultPartitionDeploymentDescriptors produced partition descriptors
+	 * @param outputGroup shuffle specific group for output metrics
+	 * @param buffersGroup shuffle specific group for buffer metrics
+	 * @return array of the task's {@link ResultPartitionWriter}s
+	 */
+	ResultPartitionWriter[] createResultPartitionWriters(
+		String taskName,
+		ExecutionAttemptID executionAttemptID,
+		Collection<ResultPartitionDeploymentDescriptor> resultPartitionDeploymentDescriptors,
+		MetricGroup outputGroup,
+		MetricGroup buffersGroup);
+
+	/**
+	 * Batch release local resources occupied with the given partitions.
+	 *
+	 * @param partitionIds partitions ids to release
+	 */
+	void releasePartitions(Collection<ResultPartitionID> partitionIds);
+
+	/**
+	 * Report unreleased partitions.
+	 *
+	 * @return collection of partitions which still occupy some resources locally on this task executor
+	 * and have been not released yet.
+	 */
+	Collection<ResultPartitionID> getUnreleasedPartitions();
+
+	/**
+	 * Factory method for the task's {@link InputGate}s.
+	 *
+	 * @param taskName string name of the task used for logs
+	 * @param executionAttemptID execution attempt id of the task
+	 * @param partitionProducerStateProvider producer state provider to query whether the producer is ready for consumption
+	 * @param inputGateDeploymentDescriptors consumed gates descriptors
+	 * @param parentGroup parent of shuffle specific metric group
+	 * @param inputGroup shuffle specific group for input metrics
+	 * @param buffersGroup shuffle specific group for buffer metrics
+	 * @return array of the task's {@link InputGate}s
+	 */
+	InputGate[] createInputGates(
+		String taskName,
+		ExecutionAttemptID executionAttemptID,
+		PartitionProducerStateProvider partitionProducerStateProvider,
+		Collection<InputGateDeploymentDescriptor> inputGateDeploymentDescriptors,
+		MetricGroup parentGroup,
+		MetricGroup inputGroup,
+		MetricGroup buffersGroup);
+
+	/**
+	 * Update consuming gate with newly available partition, previously unknown.
+	 *
+	 * @param consumerID execution id of consumer to identify belonging to it gate.
+	 * @param partitionInfo partition information to be able to consume it, e.g. network location
```

Comment:
to consume it -> to be consumed? Or `partition information for updating`

Reply:
I will change it a bit but I wanted to mention that the info enables partition consumption. I think it is already obvious that it is for the update